help_test.go

```diff
@@ -1714,3 +1712,65 @@ GLOBAL OPTIONS:
 			output.String(), expected)
 	}
 }
+
+func Test_checkShellCompleteFlag(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		name                string
+		app                 *App
+		arguments           []string
+		wantShellCompletion bool
+		wantArgs            []string
+	}{
+		{
+			name:                "disable bash completion",
+			arguments:           []string{"--generate-bash-completion"},
+			app:                 &App{},
+			wantShellCompletion: false,
+			wantArgs:            []string{"--generate-bash-completion"},
+		},
+		{
+			name:      "--generate-bash-completion isn't used",
+			arguments: []string{"foo"},
+			app: &App{
+				EnableBashCompletion: true,
+			},
+			wantShellCompletion: false,
+			wantArgs:            []string{"foo"},
+		},
+		{
+			name:      "arguments include double dash",
+			arguments: []string{"--", "foo", "--generate-bash-completion"},
+			app: &App{
+				EnableBashCompletion: true,
+			},
+			wantShellCompletion: false,
+			wantArgs:            []string{"--", "foo", "--generate-bash-completion"},
+		},
+		{
+			name:      "--generate-bash-completion",
+			arguments: []string{"foo", "--generate-bash-completion"},
+			app: &App{
+				EnableBashCompletion: true,
+			},
+			wantShellCompletion: true,
+			wantArgs:            []string{"foo"},
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt
```

Comment:
is this needed ?

Reply:
If we remove this, `go vet` fails.

```console
$ go vet ./...
# github.com/urfave/cli/v2
# [github.com/urfave/cli/v2]
./help_test.go:1765:52: loop variable tt captured by func literal
./help_test.go:1765:60: loop variable tt captured by func literal
./help_test.go:1766:7: loop variable tt captured by func literal
./help_test.go:1768:23: loop variable tt captured by func literal
./help_test.go:1770:26: loop variable tt captured by func literal
./help_test.go:1772:12: loop variable tt captured by func literal

$ echo $?
1
```

If we update go directive to 1.22 or later, we can remove this.

https://tip.golang.org/doc/go1.22

https://github.com/urfave/cli/blob/8e2384caa9350d1cd514808821c181f20afd48f0/go.mod#L3

But I don't think this change is the scope of this pull request, so I didn't do that.